### PR TITLE
Feature/namui text input

### PR DESCRIPTION
**Pull request by namse**
# What i did
- Finish making text input which only support single click
  - single click = when you double or triple click the text input, it select multiple texts, i didn't implement that yet i mean.
  - utf-8 support. Korean input tested.
- Make text input sample app in namui
- support Shift+click selection
- full support of text input by keyboard, including ctrl+arrow etc.

### CODE REVIEW

**Comment by namse** [comment]
Waiting for https://github.com/NamseEnt/namseent/pull/186

File: namui/src/namui/mod.rs [hunk 1]
```diff
@@ -93,20 +91,38 @@ pub async fn start<TProps>(
                 }
             }
             Some(NamuiEvent::MouseDown(raw_mouse_event)) => {
+                {
+                    let managers = managers();
+                    managers
+                        .text_input_manager
+                        .on_mouse_down(&namui_context, &raw_mouse_event);
+                }
```
Comment (namse) [inline namui/src/namui/mod.rs:99]: I will clean up this pattern later. I'm thinking the strategy about this.

**Comment by namse** [comment]
I found big bug. After this, button click doesn't work.

**Comment by namse** [comment]
Bug fixed in other PR. not related in this PR.

File: namui/src/namui/render/text_input/get_selection_on_mouse_down.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        let font = font.unwrap();
+
+        // const continouslyFastClickCount: number;
+
+        // if (continouslyFastClickCount >= 3) {
+        //   return getMoreTripleClickSelection({ text });
+        // }
+        // if (continouslyFastClickCount === 2) {
+        //   return getDoubleClickSelection({ text, font, x: localX });
+        // }
```
Comment (bigfoodK) [inline namui/src/namui/render/text_input/get_selection_on_mouse_down.rs:34]: What is this comment for?

Comment (bigfoodK) [reply namui/src/namui/render/text_input/get_selection_on_mouse_down.rs:34]: Aaah, I believe this will be used in the future

**Review by bigfoodK — APPROVED** [review]

**Comment by bigfoodK** [comment]
Check this pr's name before merge!

**Comment by namse** [comment]
> Check this pr's name before merge!

I thought it's enough haha

File: namui/src/namui/render/text_input/get_selection_on_mouse_down.rs (hunk 2 above)
Comment (namse) [reply namui/src/namui/render/text_input/get_selection_on_mouse_down.rs:34]: that's right, the code from past typescript code.

**Pull request merged**